src/main/java/com/artipie/asto/Key.java

```diff
@@ -106,11 +113,16 @@ public From(final Key base, final String... parts) {
          * @param parts Parts
          */
         public From(final Collection<String> parts) {
-            this.parts = Collections.unmodifiableCollection(new ArrayList<>(parts));
+            this.parts = List.copyOf(parts);
         }
 
         @Override
         public String string() {
+            for (final String part : this.parts) {
+                if (part.contains("/")) {
```

Comment:
@olegmoz also I'd suggest to extract `/` into private constant, it's used 3 times in this class. Also it's odd that qulice is not complaining about duplicated literal...

Reply:
@olenagerasimova thank you, extracted the constant. It seems that qualice ignores one symbol constants